src/app/templates/ChainSelect/ChainsDropdown.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { memo, useCallback, useMemo } from 'react';
+
+import clsx from 'clsx';
+
+import DropdownWrapper from 'app/atoms/DropdownWrapper';
+import { useShortcutAccountSelectModalIsOpened } from 'app/hooks/use-account-select-shortcut';
+import { ReactComponent as SignalAltIcon } from 'app/icons/signal-alt.svg';
+import { T } from 'lib/i18n';
+import { PopperRenderProps } from 'lib/ui/Popper';
+import { TezosChain, EvmChain, useEnabledTezosChains, useEnabledEvmChains } from 'temple/front';
+import { TempleChainTitle } from 'temple/types';
+
+import { ChainButton } from './ChainButton';
+import { ChainSelectController } from './controller';
+import styles from './style.module.css';
+
+interface Props extends PopperRenderProps {
+  controller: ChainSelectController;
+}
+
+export const ChainsDropdown = memo<Props>(({ opened, setOpened, controller }) => {
+  const selectedChain = controller.value;
+
+  const tezosChains = useEnabledTezosChains();
+  const evmChains = useEnabledEvmChains();
+
+  useShortcutAccountSelectModalIsOpened(() => setOpened(false));
+
+  const handleTezosNetworkSelect = useCallback(
+    (network: TezosChain) => {
+      controller.setValue(network);
+    },
+    [controller]
+  );
+
+  const handleEvmNetworkSelect = useCallback(
+    (network: EvmChain) => {
+      controller.setValue(network);
+    },
+    [controller]
+  );
+
+  const h2ClassName = useMemo(
+    () =>
+      clsx(
+        'flex items-center mb-2 px-1 pb-1',
+        'border-b border-white border-opacity-25',
+        'text-white text-opacity-90 text-sm text-center'
+      ),
+    []
+  );
+
+  return (
+    <DropdownWrapper opened={opened} design="dark" className="origin-top-right p-2">
+      <div className={styles.scroll}>
+        <h2 className={h2ClassName}>
+          <SignalAltIcon className="w-auto h-4 mr-1 stroke-current" />
+          {TempleChainTitle.tezos} <T id="networks" />
+        </h2>
+
+        {tezosChains.map(chain => {
```

Comment:
Please consider using a separate memoized component for these buttons and these: https://github.com/madfish-solutions/templewallet-extension/pull/1120/files#diff-ed2c115507fc19ca3989c073ed89cd1b74bdbb3c13dba393eb39afba9dec8812R84